cmake/kokkos_arch.cmake

```diff
@@ -1255,6 +1279,10 @@ foreach(ARCH IN LISTS SUPPORTED_AMD_ARCHS)
   endif()
 endforeach()
 
+if(KOKKOS_ARCH_AMD_GFX942_APU)
+  set(KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY ON)
```

Comment:
does this need to be 'Kokkos_ENABLE_IMPL_HIP_UNIFIED_MEMORY'? Also I think we could just remove the cmake level option for this and define it inside the source code based on the architecture macro. Thoughts?

Reply:
I thought we wanted to keep the CMake option for the next release and then remove. I am fine removing the CMake option now.

Reply:
Yeah let's remove it.